proto-public/pbauth/v1alpha1/traffic_permission.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package hashicorp.consul.auth.v1alpha1;
+
+message TrafficPermission {
+  Destination destination = 1;
+  Action action = 2;
+  repeated Permission permissions = 3;
+}
+
+message PartitionTrafficPermission {
+  Action action = 1;
+  repeated Permission permissions = 2;
+}
+
+message Destination {
+  string identity_name = 1;
+}
+
+enum Action {
+  ACTION_UNSPECIFIED = 0;
+  ACTION_DENY = 1;
+  ACTION_ALLOW = 2;
+}
+
+message Permission {
+  repeated Source sources = 1;
+  repeated DestinationRule destination_rules = 2;
+}
+
+message Source {
+  string identity_name = 1;
+  string namespace = 2;
+  string partition = 3;
+  string peer = 4;
+  string sameness_group = 5;
+
+  // exclude is a list of sources to exclude from this source.
+  repeated ExcludeSource exclude = 6;
+}
+
+// ExcludeSource is almost the same as source but it doesn't allow to
+// add more exclude sources.
+message ExcludeSource {
+  string identity_name = 1;
+  string namespace = 2;
+  string partition = 3;
+  string peer = 4;
+  string sameness_group = 5;
+}
+
+message DestinationRule {
+  oneof path {
+    string path_exact = 1;
+    string path_prefix = 2;
+    string path_regex = 3;
+  }
+  repeated string methods = 4;
+  DestinationRuleHeader header = 5;
+  repeated string port_names = 6;
+  repeated ExcludePermissionRule exclude = 7;
+}
+
+message ExcludePermissionRule {
+  oneof path {
+    string path_exact = 1;
+    string path_prefix = 2;
+    string path_regex = 3;
+  }
+  // methods is the list of HTTP methods. If no methods are specified,
+  // this rule will apply to all methods.
+  repeated string methods = 4;
+
+  DestinationRuleHeader header = 5;
+
+  // ports is a list of workload ports from which this permission is excluded.
+  repeated string ports = 6;
```

Comment:
I missed this on the RFC but this should probably also be `port_names` like above
```suggestion
  repeated string port_names = 6;
```